core/common/src/main/java/alluxio/security/authorization/PermissionStatus.java

```diff
@@ -86,52 +90,69 @@ public FileSystemPermission getPermission() {
    *
    * @param umask the umask to apply
    * @return a new {@link PermissionStatus}
-   * @see FileSystemPermission#applyUMask(FileSystemPermission)
    */
   public PermissionStatus applyUMask(FileSystemPermission umask) {
     FileSystemPermission newFileSystemPermission = mPermission.applyUMask(umask);
     return new PermissionStatus(mUserName, mGroupName, newFileSystemPermission);
   }
 
   /**
-   * Gets the Directory default {@link PermissionStatus}. Currently the default dir permission is
-   * 0777.
+   * Applies default umask to new created files.
    *
-   * @return the default {@link PermissionStatus} for directories
+   * @param conf the runtime configuration of Alluxio
+   * @return a new {@link PermissionStatus}
+   */
+  public PermissionStatus applyFileUMask(Configuration conf) {
+    FileSystemPermission newFileSystemPermission =
+        mPermission.applyUMask(FileSystemPermission.getUMask(conf)).applyUMask(FILE_UMASK);
+    return new PermissionStatus(mUserName, mGroupName, newFileSystemPermission);
+  }
+
+  /**
+   * Applies default umask to new created directories.
+   *
+   * @param conf the runtime configuration of Alluxio
+   * @return a new {@link PermissionStatus}
    */
-  public static PermissionStatus getDirDefault() {
-    return new PermissionStatus("", "", new FileSystemPermission(Constants
-        .DEFAULT_FS_FULL_PERMISSION));
+  public PermissionStatus applyDirectoryUMask(Configuration conf) {
+    FileSystemPermission newFileSystemPermission =
+        mPermission.applyUMask(FileSystemPermission.getUMask(conf));
+    return new PermissionStatus(mUserName, mGroupName, newFileSystemPermission);
   }
 
-  // TODO(binfan): remove remote parameter by making two different get methods
   /**
    * Creates the {@link PermissionStatus} for a file or a directory.
```

Comment:
that is not in the constructor. 0755 (for dirs) or 0644 (for files) are created by applying umask for dir and umask for file on top of default permission 0777, the logic to apply umask according to dir or file is in `InodeTree.createPath` which is the entry point for any operations in higher level that add new inode to the existing inode tree.